infobip_channels/sms/models/body/reschedule_sms_messages.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime, timedelta
+from typing import Optional, Union
+
+from pydantic import validator
+
+from infobip_channels.core.models import MessageBodyBase
+
+
+class RescheduleSMSMessagesMessageBody(MessageBodyBase):
+    send_at: Optional[Union[datetime, str]] = None
+
+    @validator("send_at")
+    def convert_send_at_to_correct_format(cls, value):
```

Comment:
This is very similar to the `convert_send_at_to_correct_format` method we use in the `mms` channel. We could extract the common logic into a Validator class and change this version to:

```python
class RescheduleSMSMessagesMessageBody(Validator, MessageBodyBase):
    ...
    @validator("send_at")
    def convert_send_at_to_correct_format(cls, value):
            if value > datetime.now() + timedelta(days=180):
                raise ValueError(
                    "Scheduled message must me sooner than 180 days from today"
                )
            return super().convert_send_at_to_correct_format(value)
```

Reply:
done done, thanks